### PR TITLE
Cleanup redundant input args

### PR DESCRIPTION
**Pull request by norberthu**
**What changed?**

**Why?**

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: common/persistence/persistenceInterface.go [hunk 1]
```diff
@@ -553,8 +551,6 @@ type (
 	InternalReadHistoryBranchRequest struct {
 		// The raw branch token
 		BranchToken []byte
-		// The tree of branch range to be read
-		TreeID string
 		// The branch range to be read
 		BranchID string
```
[the remark this answers is not in the document]
Comment (norberthu) [reply common/persistence/persistenceInterface.go:555]: My rationale is `TreeID` is also inside `BranchToken`... so if you look at each of the history store implementations, I parse the tree ID out of the branch token

[the remark this answers is not in the document]
Comment (norberthu) [reply common/persistence/persistenceInterface.go:555]: oops... haha... my eyes focused on a different line =)

good catch! `BranchID` can also be removed! I will remove it

[the remark this answers is not in the document]
Comment (norberthu) [reply common/persistence/persistenceInterface.go:555]: oops... haha... my eyes focused on a different line =)

Short answer is, I don't believe we can remove it. Let's see you can follow my reasoning below:

- judging from the existing code usage of this field, `branchToken` is the current branch token, which contains the tree id, the current branch id, and the ancestors (see https://github.com/temporalio/temporal/blob/master/common/persistence/history_manager.go#L709-L720)
- the `readRawHistoryBranch()` helper is trying to paginate over the entire set of ancestors of the same tree id
- the branch id for each of ancestors are different (see https://github.com/temporalio/temporal/blob/master/proto/internal/temporal/server/api/persistence/v1/history_tree.proto#L49-L57)

[the remark this answers is not in the document]
Comment (yux0) [reply common/persistence/persistenceInterface.go:555]: In this case, could we just remove BranchToken []byte?

[the remark this answers is not in the document]
Comment (norberthu) [reply common/persistence/persistenceInterface.go:555]: Unfortunately no. Branch token may contain more info required to locate the actual data

**Comment by yux0** [comment]
Could we add some unit test/integration test coverage?

**Review by yux0 — APPROVED** [review]

**Pull request merged**